test/functional/feature_remove_pruned_files_on_startup.py

```diff
@@ -18,38 +19,48 @@ def mine_batches(self, blocks):
         for _ in range(n):
             self.generate(self.nodes[0], 250)
         self.generate(self.nodes[0], blocks % 250)
-        self.sync_blocks()
 
     def run_test(self):
         blk0 = self.nodes[0].blocks_path / "blk00000.dat"
         rev0 = self.nodes[0].blocks_path / "rev00000.dat"
         blk1 = self.nodes[0].blocks_path / "blk00001.dat"
         rev1 = self.nodes[0].blocks_path / "rev00001.dat"
         self.mine_batches(800)
-        fo1 = os.open(blk0, os.O_RDONLY)
-        fo2 = os.open(rev1, os.O_RDONLY)
-        fd1 = os.fdopen(fo1)
-        fd2 = os.fdopen(fo2)
+
+        self.log.info("Open some files to check that this may delay deletion")
+        fd1 = open(blk0, "rb")
+        fd2 = open(rev1, "rb")
         self.nodes[0].pruneblockchain(600)
 
         # Windows systems will not remove files with an open fd
         if platform.system() != 'Windows':
-            assert not os.path.exists(blk0)
-            assert not os.path.exists(rev0)
-            assert not os.path.exists(blk1)
-            assert not os.path.exists(rev1)
+            assert not blk0.exists()
+            assert not rev0.exists()
+            assert not blk1.exists()
+            assert not rev1.exists()
         else:
-            assert os.path.exists(blk0)
-            assert not os.path.exists(rev0)
-            assert not os.path.exists(blk1)
-            assert os.path.exists(rev1)
+            assert blk0.exists()
+            assert not rev0.exists()
+            assert not blk1.exists()
+            assert rev1.exists()
 
-        # Check that the files are removed on restart once the fds are closed
+        self.log.info("Check that the files are removed on restart once the fds are closed")
         fd1.close()
         fd2.close()
         self.restart_node(0)
-        assert not os.path.exists(blk0)
-        assert not os.path.exists(rev1)
+        assert not blk0.exists()
+        assert not rev1.exists()
+
+        self.log.info("Check that a reindex will wipe all files")
```

Comment:
Might it be good to check the file count here before deletion during the reindex?